worker/uniter/operation/executor.go

```diff
@@ -70,9 +71,22 @@ func (x *executor) Run(op Operation) error {
 	switch err := x.do(op, stepPrepare); errors.Cause(err) {
 	case ErrSkipExecute:
 	case nil:
+		done := make(chan struct{})
+		go func() {
+			for {
+				select {
+				case rs := <-remoteStateChange:
+					op.RemoteStateChanged(rs)
+				case <-done:
+					return
+				}
+			}
+		}()
 		if err := x.do(op, stepExecute); err != nil {
+			close(done)
```

Comment:
defer close(done) above?

Reply:
No because I only wanted the remote state change to trigger during the execute phase of the operation.